droidlet/interpreter/craftassist/tasks.py

```diff
@@ -1067,25 +1082,35 @@ def step(self):
             agent.send_chat("I can't find it in my inventory!")
             self.finished = False
             return
+        dist_to_target = np.linalg.norm(np.array(agent.pos) - np.array(self.target))
+        close_to_target = dist_to_target <= self.approx
+        if not close_to_target:
+            move_task = Move(agent, {"target": self.target, "approx": self.approx})
+            self.add_child_task(move_task)
+            self.tries = self.tries - 1
+            return
+
         node = agent.memory.get_mem_by_id(self.obj_memid)
         count = node.count
-        id, m = self.idm
-        agent.drop_inventory_item_stack(id, m, count)
-        ItemStackNode.remove_from_inventory(agent.memory, node)
-
-        mindist, dropped_item_stack = self.get_nearby_new_item_stack(agent, id, m)
+        if getattr(agent, "backend", "cuberite") == "cuberite":
```

Comment:
oooo this should go away once we fix the integration with pyworld right ?

Reply:
probably not- getting and dropping things is different in pyworld (where you explicitly pick) than cuberite (where walking close picks it up); and cuberite tracks item stacks in a different way well